com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/validation/Validator.java

```diff
@@ -288,6 +290,26 @@ protected void checkMissingRequiredProperties(Set<SwaggerError> errors, Abstract
         }
     }
 
+    /**
+     * This method checks that referenced objects are of expected type as defined in the schema.
+     * 
+     * @param errors
+     * @param node
+     */
+    protected void checkReferenceType(Set<SwaggerError> errors, AbstractNode node) {
+        if (JsonReference.isReference(node)) {
+            Model model = node.getModel();
+            TypeDefinition type = node.getType();
+
+            AbstractNode nodeValue = node.get(JsonReference.PROPERTY);
+            AbstractNode valueNode = model.find((String) nodeValue.asValue().getValue());
```

Comment:
It only works for local references. It throws and exception for external references:
* From the same workspace (local external reference via relative path)
* Or, remove (accessed via HTTP/HTTPS/any other protocol)

Because it throws an exception which is never processed, the exception is propagated to the top level `com.reprezen.swagedit.core.validation.Validator.validate(JsonDocument, URI)` and it means that validation errors are not processed at all.